lib/Dialect/HW/InnerSymbolTable.cpp

```diff
@@ -173,15 +382,32 @@ InnerSymbolTableCollection::populateAndVerifyTables(Operation *innerRefNSOp) {
         return op->hasTrait<OpTrait::InnerSymbolTable>();
       }));
 
-  // Ensure entries exist for each operation.
-  llvm::for_each(innerSymTableOps,
-                 [&](auto *op) { symbolTables.try_emplace(op, nullptr); });
+  // Prime the top-level symbol tables.
+  bool scanFailed = false;
+  llvm::for_each(innerSymTableOps, [&](auto *op) {
```

Comment:
cc @dtzSiFive What if the InnerRefNamespace is also an InnerSymbolTable, you won't find the InnerSymbol defined under it, e.g.
```mlir
circt.container [IST, IRN] {
  ibis.container [IS, IST] { }
  }
```

We also allow InnerSymbols to be defined in nested regions:
```mlir
circt.container [IST, IRN] {
  circt.region_op {
    ibis.container [IS, IST] { }
  }
}
```